app/src/main/java/dev/smoketrees/twist/ui/home/HomeFragment.kt

```diff
@@ -48,32 +47,44 @@ class HomeFragment : BaseFragment<FragmentHomeBinding, AnimeViewModel>(
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         showLoader()
+        noticeClear()
         dataBinding.hasResult = false
 
+        // Show notice on error
+        viewModel.lastCode
+                .observe(viewLifecycleOwner, Observer {
+                    if (!viewModel.areAllLoaded.value!!) {
+                        notice(it)
+                    } //TODO: else show small bar above motd
+                })
+
+        viewModel.areAllLoaded // Only show contents when everything is loaded
+                .observe(viewLifecycleOwner, Observer { dataBinding.hasResult = it })
+
+        // Load data
         viewModel.getAllAnime()
         viewModel.getTrendingAnime(40)
 
         viewModel.trendingAnimeLiveData
-            .observe(viewLifecycleOwner, Observer { trendingList ->
-                when (trendingList.status) {
-                    Result.Status.LOADING -> {
-                        showLoader()
-                        dataBinding.hasResult = false
-                    }
+                .observe(viewLifecycleOwner, Observer { trendingList ->
+                    when (trendingList.status) {
+                        Result.Status.LOADING -> {
+                            showLoader()
+                        }
 
-                    Result.Status.SUCCESS -> {
-                        if (!trendingList.data.isNullOrEmpty()) {
-                            hideLoader()
-                            dataBinding.hasResult = true
-                            trendingAdapter.updateData(trendingList.data)
+                        Result.Status.SUCCESS -> {
+                            if (!trendingList.data.isNullOrEmpty()) {
+                                hideLoader()
```

Comment:
I'd suggest moving `hideLoader()` to the `areAllLoaded` observer since trendingList returns a result very early. Everything else LGTM.

Reply:
Moved